ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -195,8 +197,24 @@ protected BlockCreationResult createBlock(
 
       throwIfStopped();
 
-      final Optional<List<Deposit>> maybeDeposits =
-          Optional.empty(); // TODO 6110: Extract deposits from transaction receipts
+      final DepositsValidator depositsValidator = newProtocolSpec.getDepositsValidator();
+      Optional<List<Deposit>> maybeDeposits = Optional.empty();
+
+      if (depositsValidator instanceof DepositsValidator.AllowedDeposits) {
+        final List<Deposit> depositsFromReceipts =
+            transactionResults.getReceipts().stream()
+                .flatMap(receipt -> receipt.getLogsList().stream())
+                .filter(
+                    log ->
+                        ((DepositsValidator.AllowedDeposits) depositsValidator)
+                            .getDepositContractAddress()
```

Comment:
Doesn't seem quite to be exposing a getter on the DepositValidator. Would rather we pass in the depositContractAddress from the config.

Reply:
> Doesn't seem quite right to be exposing a getter on the DepositValidator. Would rather we pass in the depositContractAddress from the config.

Yea I thought the same. However I am not sure how to do it without making a big change. Should I add the `depositContractAddress` as a param of `createBlock()`? If so, seems like I would need to update all the callers of `createBlock()`. Would be great if you could give me a pointer.

Reply:
Can't think of a way that doesn't end up changing at 10 or so files. 

I was thinking of passing it into the constructor of the `AbstractBlockCreator`, which would also mean passing it into the `MergeBlockCreator` and `MergeCoordinator` and constructing the `MergeCoordinator` using the value of the deposit contract from the genesis in the `MergeBesuControllerBuilder`. Still quite a few changes. There are other BlockCreator instances the Pow, Bft but they could be instantiated with an Optional.empty for the deposit contract.